### PR TITLE
Fix ddp_spawn -> ddp fallback logic when on LSF cluster

### LINKED ISSUE #15103: DDP-spawn on LSF cluster (multi-node) should not be supported
## 🚀 Bug

DDP-spawn on multi-node configurations where processes get spawned externally should not be supported.

### To Reproduce

Running with

```py
Trainer(strategy="ddp_spawn", num_nodes>1, devices=...)
```

on LSF is currently faulty but should not be supported. 

### Expected behavior

Should not be supported. Two options:

A) Raise an error and suggest to use `strategy="ddp"`.
B) Fall-back to DDP. A similar check already exists for SLURM:

https://github.com/Lightning-AI/lightning/blob/d2840a20bd27a0a86fc59ba54d94543e63523133/src/pytorch_lightning/trainer/connectors/accelerator_connector.py#L616-L619

I suggest to go with B) as we already do it with the other environments. 

### Additional context

Originally reported on [Slack](https://pytorch-lightning.slack.com/archives/C0178DAJSQM/p1665520147457089).

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/Lightning-AI/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/Lightning-AI/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/Lightning-AI/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/Lightning-AI/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging PyTorch Lightning, Transformers, and Hydra.

### PR DESCRIPTION
**Pull request by Atharva-Phatak**
## What does this PR do?

Implements the fix proposed in #15103 

Fixes #15103 

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Atharva-Phatak** [comment]
@awaelchli I am not able to figureout one environment variable for the LSF specifically the rank file. Would like some guidance on it, if possible.

**Comment by awaelchli** [comment]
Great. Pushed an update :)

**Review by awaelchli — APPROVED** [review]

**Review by carmocca — APPROVED** [review]
